src/puppetlabs/trapperkeeper/authorization/acl.clj

```diff
@@ -56,123 +82,165 @@
       (clojure.string/split #"\.")
       reverse))
 
-(schema/defn new-domain :- Entry
+(schema/defn ^:always-validate new-domain :- ACE
   "Creates a new ACE for a domain"
-  [type :- AuthType
-   pattern :- schema/Str]
+  [auth-type :- AuthType
+   {:keys [certname extensions]} :- ACEConfig]
   (cond
+    ;; CSR Attributes
```

Comment:
nit - should just refer to "extensions" here rather than CSR attributes, even though I suppose "extensions" would be obvious from the following code?